cli/src/execDocker/docker.js

```diff
@@ -109,15 +127,24 @@ export async function pushDockerImage({
     throw new Error('Missing DockerHub credentials.');
   }
   const dockerImage = docker.getImage(tag);
+  const abortSignal = createAbortSignal();
 
   const imagePushStream = await dockerImage.push({
     authconfig: {
       username: dockerhubUsername,
       password: dockerhubAccessToken,
     },
+    abortSignal,
   });
-
   await new Promise((resolve, reject) => {
+    // Handle abort signal
+    if (abortSignal) {
+      abortSignal.addEventListener('abort', () => {
+        imagePushStream.destroy();
+        reject(new Error('Docker push aborted'));
+      });
+    }
+
```

Comment:
same as for build, `dockerImage.push({ abortSignal })` already takes care of stopping the push and calling `onFinished` with an error, no need to handle it manually.